docs/src/cookbooks/two-side-column.md

```diff
@@ -0,0 +1,749 @@
+```
+</td>
+         </tr>
+         <tr>
+<td>
+
+```
+module ParameterizedWidthAdder(
+  input [in0Width-1:0] in0,
+  input [in1Width-1:0] in1,
+  output [sumWidth-1:0] sum
+);
+  parameter in0Width = 8;
+  parameter in1Width = 1;
+  parameter sumWidth = 9;
+
+  assign sum = in0 + in1;
+
+endmodule
+```
+
+</td>
+<td>
+
+```
+class ParameterizedWidthAdder(in0Width: Int, in1Width: Int, sumWidth: Int) extends Tester{
+  require(in0Width >= 0)
+  require(in1Width >= 0)
+  require(sumWidth >= 0)
+  val io = IO(new Bundle {
+    val in0 = Input(UInt(in0Width.W))
+    val in1 = Input(UInt(in1Width.W))
+    val sum = Output(UInt(sumWidth.W))
+  })
+  // a +& b includes the carry, a + b does not
+  io.sum := io.in0 +& io.in1
+}
+```
+</td>
+</tr>
+<tr>
+<td>
+
+```
+module TestBench;
+wire [31:0] sum;
+ParameterizedWidthAdder  #(32, 32, 32) my32BitAdderWithTruncation (32'b0, 32'b0, sum);
+endmodule
+```
+</td>
+            <td>
+
+```
+val my32BitAdderWithTruncation = Module(new ParameterizedWidthAdder(32, 32, 32)
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+# Wire assignment
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```
+wire [31:0] 
+a = 32'd42; 
+wire [31:0] 
+b = 32'hbabecafe; 
+wire [15:0] c; 
+assign c = 16'b1;
+```
+
+</td>
+<td>
+
+```scala mdoc:silent
+
+
+class MyWireAssignmentModule extends Module {
+ val a = WireDefault(42.U(32.W))
+ val b = WireDefault("hbabecafe".U(32.W))  
+ val c = Wire(UInt(16.W)) 
+  c := "b1".U;
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new MyWireAssignmentModule)
+```
+</td>
+         </tr>
+<tr>
+<td>
+
+```
+module MyWireAssignmentModule(
+  input   clock,
+  input   reset
+);
+endmodule
+```
+
+
+</td>
+<td>
+
+```scala mdoc:silent
+class MyWireAssignmentModule2 extends Module {
+ val a = WireDefault(42.U(32.W))
+ val aa = 42.U(32.W)
+ val b = WireDefault("hbabecafe".U(32.W))  
+ val c = Wire(UInt(16.W)) 
+ val d = Wire(Bool())
+ val e = Wire(UInt())
+ val f = WireDefault("hdead".U)
+ val g = Wire(SInt(64.W))
+ val h = WireDefault(5.asSInt(32.W))
+ val i = WireDefault((3.S(16.W)))
+
+c := "b1".U;
+d := true.B
+e := "b1".U(3.W);
+g := -5.S
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new MyWireAssignmentModule2)
+```
+
+</td>
+</tr>
+    </table>
+<html>
+<body>
+
+# Register assignment
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```
+module RegisterModule(
+  input         clock,
+  input         reset,
+  input  [11:0] io_in,
+  output [11:0] io_out
+);
+  assign io_out = io_in; // @[main.scala 18:10]
+endmodule
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class RegisterModule extends Module {
```

Comment:
Here is a more "interesting" RegisterModule Scastie:

https://scastie.scala-lang.org/N5X57mPcQWWYbRrqs4YWkg

Reply:
And here is the Human-Written Verilog:

```
module RegisterModule(
  input        clock,
  input        reset,
  input  [7:0] in,
  output [7:0] out,
  input        differentClock,
  input        differentSyncReset,
  input        differentAsyncReset
);

reg [7:0] registerWithoutInit;
reg [7:0] registerWithInit;
reg [7:0] registerOnDifferentClockAndSyncReset;
reg [7:0] registerOnDifferentClockAndAsyncReset;


 always @(posedge clock) begin
    registerWithoutInit <= in + 8'h1;
 end

 always @(posedge clock) begin
    if (reset) begin
      registerWithInit <= 8'd42;
    end else begin
      registerWithInit <= registerWithInit - 8'h1;
    end
  end
 
 always @(posedge differentClock) begin
    if (differentSyncReset) begin
      registerOnDifferentClockAndSyncReset <= 8'h42;
    end else begin
      registerOnDifferentClockAndSyncReset <= in - 8'h1;
    end
 end
  
 always @(posedge differentClock or posedge differentAsyncReset) begin
   if (differentAsyncReset) begin
      registerOnDifferentClockAndAsyncReset <= 8'h24;
   end else begin
      registerOnDifferentClockAndAsyncReset <= in + 8'h2;
   end
 end

  assign out = in + registerWithoutInit + registerWithInit + registerOnDifferentClockAndSyncReset registerOnDifferentClockAndAsyncReset;


endmodule
```